instrumentation/action_pack/lib/opentelemetry/instrumentation/action_pack/patches/action_controller/metal.rb

```diff
@@ -14,18 +14,12 @@ module Metal
             def dispatch(name, request, response)
               rack_span = OpenTelemetry::Instrumentation::Rack.current_span
               if rack_span.recording?
-                unless request.env['action_dispatch.exception']
-                  rack_span.name = case instrumentation_config[:span_naming]
-                                   when :controller_action then "#{self.class.name}##{name}"
-                                   else "#{request.method} #{rails_route(request)}"
-                                   end
-                end
+                rack_span.name = "#{self.class.name}##{name}" unless request.env['action_dispatch.exception']
```

Comment:
In days of yore, it looks like we [checked for valid context on the rack span](https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/51b6e8ad465ba81fa743224b5ecfd4ab871bea51/instrumentation/action_pack/lib/opentelemetry/instrumentation/action_pack/patches/action_controller/metal.rb#L16). Not sure if we want to bring that back; not a blocker, just a detail I wanted to surface.

Reply:
@plantfansam I may be incorrect here but I think that checking if it is read/write span i.e. `recording?`, is better than checking for a valid context. 

If I understand this code correctly, it is possible to have a read-only span that has a valid context: 

https://github.com/open-telemetry/opentelemetry-ruby/blob/a4efaf3a44b0efc6ca805a6d0a5bdc4f13a01b34/sdk/lib/opentelemetry/sdk/trace/tracer_provider.rb#LL161C81-L161C81

I think that means it would call the no-opt `name` and `add_attributes` methods: https://github.com/open-telemetry/opentelemetry-ruby/blob/a4efaf3a44b0efc6ca805a6d0a5bdc4f13a01b34/api/lib/opentelemetry/trace/span.rb#L138

Reply:
I read the code the same way @arielvalentin